server/src/main/java/com/flipkart/grayskull/mappers/SecretMapper.java

```diff
@@ -0,0 +1,102 @@
+package com.flipkart.grayskull.mappers;
+
+import com.flipkart.grayskull.models.db.Secret;
+import com.flipkart.grayskull.models.db.SecretData;
+import com.flipkart.grayskull.models.dto.request.CreateSecretRequest;
+import com.flipkart.grayskull.models.dto.request.UpgradeSecretDataRequest;
+import com.flipkart.grayskull.models.dto.response.CreateSecretResponse;
+import com.flipkart.grayskull.models.dto.response.SecretDataResponse;
+import com.flipkart.grayskull.models.dto.response.SecretDataVersionResponse;
+import com.flipkart.grayskull.models.dto.response.SecretMetadata;
+import com.flipkart.grayskull.models.enums.SecretState;
+import org.mapstruct.InheritConfiguration;
+import org.mapstruct.Mapper;
+import org.mapstruct.Mapping;
+import org.mapstruct.ReportingPolicy;
+
+import java.time.Instant;
+import java.util.UUID;
+
+/**
+ * Maps between Secret domain models (entities) and Data Transfer Objects (DTOs).
+ * <p>
+ * This mapper uses MapStruct to generate the implementation at compile-time,
+ * ensuring high performance and type safety.
+ */
+@Mapper(componentModel = "spring", unmappedTargetPolicy = ReportingPolicy.IGNORE,
+        imports = {UUID.class, Instant.class, SecretState.class})
+public interface SecretMapper {
```

Comment:
is this class tested? please add tests for each type of mapping with various inputs .. null, malformed, partial null..